navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -260,6 +309,38 @@ private static ObjectNode putInstruction(PointList points, InstructionList instr
         Instruction instruction = instructions.get(instructionIndex);
         ArrayNode intersections = instructionJson.putArray("intersections");
 
+        // make pointList writeable
+        PointList pointList = instruction.getPoints().clone(false);
+
+        if (instructionIndex == 0) {
+             // Fix first (departure) instruction
+             // to only have one "out" intersection
+            intersectionDetails = fixFirstIntersectionDetail(intersectionDetails);
```

Comment:
I am wondering why we are doing this here and not in line 111,  in putRouteInformation? I think that is the place where I would have expected this to be. I understand that you don't want to change the original intersection details but this is only used in the putInstruction right now anyways?

Reply:
Yeah, moving it up and modifying the Map on the fly makes the code easier.